File: ng-resume/src/app/resume/models/workhistory.ts
/**
 * Interface to define structure of a work history item object.
 */
export interface WorkHistory {
    id: string;
    startYear: number;
    endYear: number;
    title: string;
    organization: string;
    organizationURL?: URL;
    bullets: string[];
    skills: string[];
}

File: ng-resume-api/models/WorkHistory.cs
using System;
using Newtonsoft.Json;

namespace Jpf.NgResume.Api.Models
{
    /// <summary>
    /// Model class to represent work history items.
    /// </summary>
    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    public class WorkHistory
    {
        public Guid Id {get; set;}
        public int StartYear {get; set;}
        public int EndYear {get; set;}
        public string Title {get; set;}
        public string Organization {get; set;}
        public string OrganizationURL {get; set;}
        public string[] Bullets {get; set;}
        public string[] Skills {get; set;}
    }
}
